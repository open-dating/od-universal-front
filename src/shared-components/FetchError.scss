.fetch-error {
  margin-bottom: 16px;
}
