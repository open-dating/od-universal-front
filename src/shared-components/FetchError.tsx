import React from 'react'
import { Alert, AlertTitle } from '@material-ui/lab'

import './FetchError.scss'
import {recognizeError} from '../utils/errorHelpers'

export function FetchError({error}: {error: any}) {
  const message = recognizeError(error)

  return (
    <Alert
      className="fetch-error"
      severity="error"
    >
      <AlertTitle>Error</AlertTitle>
      {message}
    </Alert>
  )
}
