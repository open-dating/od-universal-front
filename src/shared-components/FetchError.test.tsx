import React from 'react'
import { shallow } from 'enzyme'
import {FetchError} from './FetchError'

describe('FetchError', () => {
    it('toMatchSnaphoot', () => {
        const wrapper = shallow(<FetchError error='Errror message'></FetchError>)
        expect(wrapper).toMatchSnapshot()
    })
})