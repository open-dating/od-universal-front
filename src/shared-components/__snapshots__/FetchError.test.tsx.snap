// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchError toMatchSnaphoot 1`] = `
<WithStyles(ForwardRef(Alert))
  className="fetch-error"
  severity="error"
>
  <WithStyles(ForwardRef(AlertTitle))>
    sharedComponents.error
  </WithStyles(ForwardRef(AlertTitle))>
  Errror message
</WithStyles(ForwardRef(Alert))>
`;
